File: java/Sort.java
// required modules
import java.util.Arrays;

public class Sort {
  // get console params
  public static void main(String[] inputArray) {
    Float[] floatInputArray = Helpers.toFloatArray(inputArray);
    Float[] sortedArray = Sort.perform(floatInputArray);
    // test
    System.out.println(Helpers.outputResult(inputArray, Helpers.toStringArray(sortedArray)));
  }

  // development
  private static Float[] perform(Float[] array) {
    boolean hasChanges = true;
    while (hasChanges) {
      hasChanges = false;

      for (int i = 0; i < array.length - 1; i++) {
        if (array[i] > array[i + 1]) {
          hasChanges = true;
          float temp = array[i];
          array[i] = array[i + 1];
          array[i + 1] = temp;
        }
      }
    }

    return array;
  }
}

// custom helper functions
class Helpers {
  public static Float[] toFloatArray(String[] array) {
    return Arrays.stream(array)
      .map(Float::valueOf)
      .toArray(Float[]::new);
  }

  public static String[] toStringArray(Float[] array) {
    return Arrays.stream(array)
      .map(Helpers::convertNumberToString)
      .toArray(String[]::new);
  }

  public static String convertNumberToString(Float number) {
    return isInteger(number) ?
      String.valueOf(Math.round(number)) :
      String.valueOf(number);
  }

  public static Boolean isInteger(Float number) {
    return number % 1 == 0;
  }

  public static String outputResult(String[] inputArray, String[] outputArray) {
    return ""
      .concat(arrayToString(inputArray))
      .concat(" -> ")
      .concat(arrayToString(outputArray));
  }

  public static String arrayToString(String[] array) {
    return ""
      .concat("[")
      .concat(String.join(", ", array))
      .concat("]");
  }
}

File: kotlin/sort.kt
fun main(args: Array<String>) {
  val inputArray = args.toList().map { it.toFloat() };
  val sortedArray = sort(inputArray);
  println(arrayToString(inputArray) + " -> " + arrayToString(sortedArray))
}

fun sort(array: List<Float>): List<Float> {
  var hasChanges = true
  var sortedArray = array.toMutableList()
  while (hasChanges) {
    hasChanges = false

    for (i in 0 until sortedArray.size - 1) {
      if (sortedArray[i] > sortedArray[i + 1]) {
        hasChanges = true
        val temp = sortedArray[i]
        sortedArray[i] = sortedArray[i + 1]
        sortedArray[i + 1] = temp
      }
    }
  }

  return sortedArray
}

fun arrayToString(array: List<Float>): String {
  return array
    .map { if (isInt(it)) it.toInt().toString() else it.toString() }
    .joinToString(prefix = "[", postfix = "]")
}

fun isInt(number: Float): Boolean {
  return number.toInt().toFloat().equals(number)
}
